src/fromager/bootstrapper.py

```diff
@@ -56,9 +58,9 @@ def __init__(
         # the dependency list. The key is the requirements spec, rather
         # than the package, in case we do have multiple rules for the same
         # package.
-        self._seen_requirements: set[tuple[NormalizedName, tuple[str, ...], str]] = (
-            set()
-        )
+        # sdists and wheels are tracked separately for sdist_only mode.
```

Comment:
It wasn't immediately clear why we needed to track them separately. I think it's because if something shows up as a build dependency then we have to build the wheel, even if we have seen it before and used the sdist approach. Could you add that to this comment, if I'm correct?

I'm also curious about why adding the sdist_only flag to the search key wasn't enough to determine if we had seen a package, instead of having separate sets.

Reply:
Oh, that is a clever idea. Let me change the code and add a longer explanation.

Reply:
I have changed the cache key to:

```python
SeenKey = tuple[NormalizedName, tuple[str, ...], str, typing.Literal["sdist", "wheel"]]
```